library/std/src/io/buffered/bufwriter.rs

```diff
@@ -339,7 +421,13 @@ impl<W: Write> Write for BufWriter<W> {
                 self.panicked = false;
                 r
             } else {
-                bufs.iter().for_each(|b| self.buf.extend_from_slice(b));
+                // SAFETY: We checked whether or not the spare capacity was large enough above. If
+                // it was, then we're safe already. If it wasn't, we flushed, making sufficient
+                // room for any input <= the buffer size, which includes this input.
+                unsafe {
+                    bufs.iter().for_each(|b| self.write_to_buffer_unchecked(b));
+                };
```

Comment:
```suggestion
                bufs.iter().for_each(|b| unsafe { self.write_to_buffer_unchecked(b) });
```
Would this be better?

Reply:
I don't have a big preference one way or the other. :shrug:

Reply:
Well, I tried it and didn't like the way it looked. Thanks, though. :)